duckduckgo/iOS
# Add background refresh stat safeguards

Task/Issue URL: https://app.asana.com/0/1199333091098016/1199644311779247
CC: @bwaresiak 

**Description**:

This PR introduces a mechanism which checks whether a background task's completion handler has been called previously, and corrects the background refresh metrics if so.

**Steps to test this PR**:
1. Test the background refresh flow on a device through Xcode and verify that the typical flows work as expected; there are instructions inside `AppConfigurationFetch` for manually triggering background tasks

**Device Testing**:


**OS Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
